src/main/java/cash/z/ecc/android/sdk/tool/SaplingParamTool.kt

```diff
@@ -0,0 +1,135 @@
+                    }
+                } catch (e: Throwable) {
+                    twig("failed to fetch params due to: $e")
+                    throw TransactionEncoderException.MissingParamsException
+                }
+            }
+        }
+
+        /**
+         * Download and store the params into the given directory.
+         *
+         * @param destinationDir the directory where the params will be stored. It's assumed that we
+         * have write access to this directory. Typically, this should be the app's cache directory
+         * because it is not harmful if these files are cleared by the user since they are downloaded
+         * on-demand.
+         */
+        suspend fun fetchParams(destinationDir: String) = withContext(Dispatchers.IO) {
+            val client = createHttpClient()
+            var failureMessage = ""
+            arrayOf(
+                ZcashSdk.SPEND_PARAM_FILE_NAME,
+                ZcashSdk.OUTPUT_PARAM_FILE_NAME
+            ).forEach { paramFileName ->
+                val url = "${ZcashSdk.CLOUD_PARAM_DIR_URL}/$paramFileName"
+                val request = Request.Builder().url(url).build()
+                val response = client.newCall(request).execute()
+                if (response.isSuccessful) {
+                    twig("fetch succeeded")
+                    val file = File(destinationDir, paramFileName)
+                    if (file.parentFile.exists()) {
+                        twig("directory exists!")
+                    } else {
+                        twig("directory did not exist attempting to make it")
+                        file.parentFile.mkdirs()
+                    }
+                    Okio.buffer(Okio.sink(file)).use {
+                        twig("writing to $file")
+                        it.writeAll(response.body().source())
+                    }
+                    twig("fetch succeeded, done writing $paramFileName")
+                } else {
+                    failureMessage += "Error while fetching $paramFileName : $response\n"
+                    twig(failureMessage)
+                }
+            }
+            if (failureMessage.isNotEmpty()) throw TransactionEncoderException.FetchParamsException(
+                failureMessage
+            )
+        }
+
+
+        fun clear(destinationDir: String){
+            if(validate(destinationDir)){
+                arrayOf(
+                    ZcashSdk.SPEND_PARAM_FILE_NAME,
+                    ZcashSdk.OUTPUT_PARAM_FILE_NAME
+                ).forEach { paramFileName ->
+                    val file = File(destinationDir, paramFileName)
+                    if(file.deleteRecursively()){
+                        twig("Files deleted successfully")
+                    }else{
+                        twig("Error: Files not able to be deleted!")
+                    }
+                }
+            }
+        }
+
+        fun validate(destinationDir: String): Boolean{
```

Comment:
if no further validations will be added I would consider renaming the function to express clear intent of what's being validated (existence, for now)

Reply:
Good point. Additional validations will be added in https://github.com/zcash/zcash-android-wallet-sdk/issues/205